crates/ethereum/primitives/src/alloy_compat.rs

```diff
@@ -41,3 +42,41 @@ impl TryFrom<AnyRpcTransaction> for TransactionSigned {
         Ok(Self::new(transaction, signature, hash))
     }
 }
+
+impl<T> TryFrom<AlloyRpcTransaction<T>> for TransactionSigned
+where
+    T: Into<AnyTxEnvelope>,
```

Comment:
I think I miscommunicated this in the issue we want

```suggestion
where
    T: Into<TransactionSigned>,
```

here so that the impl just becomes `tx.inner.into()` and `tx.inner.try_into()`

we want both a `From` and a `TryFrom` impl